garage/np/algos/batch_polopt.py

```diff
@@ -115,6 +118,155 @@ def log_diagnostics(self, paths):
+                observations=observations,
+                actions=actions,
+                rewards=rewards,
+                returns=returns,
+                advantages=advantages,
+                env_infos=env_infos,
+                agent_infos=agent_infos,
+                paths=paths,
+            )
+        else:
+            max_path_length = max([len(path['advantages']) for path in paths])
+
+            # make all paths the same length (pad extra advantages with 0)
+            obs = [path['observations'] for path in paths]
+            obs = tensor_utils.pad_tensor_n(obs, max_path_length)
+
+            if self.center_adv:
+                raw_adv = np.concatenate(
+                    [path['advantages'] for path in paths])
+                adv_mean = np.mean(raw_adv)
+                adv_std = np.std(raw_adv) + 1e-8
+                adv = [(path['advantages'] - adv_mean) / adv_std
+                       for path in paths]
+            else:
+                adv = [path['advantages'] for path in paths]
+
+            adv = np.asarray(
+                [tensor_utils.pad_tensor(a, max_path_length) for a in adv])
+
+            actions = [path['actions'] for path in paths]
+            actions = tensor_utils.pad_tensor_n(actions, max_path_length)
+
+            rewards = [path['rewards'] for path in paths]
+            rewards = tensor_utils.pad_tensor_n(rewards, max_path_length)
+
+            returns = [path['returns'] for path in paths]
+            returns = tensor_utils.pad_tensor_n(returns, max_path_length)
+
+            agent_infos = [path['agent_infos'] for path in paths]
+            agent_infos = tensor_utils.stack_tensor_dict_list([
+                tensor_utils.pad_tensor_dict(p, max_path_length)
+                for p in agent_infos
+            ])
+
+            env_infos = [path['env_infos'] for path in paths]
+            env_infos = tensor_utils.stack_tensor_dict_list([
+                tensor_utils.pad_tensor_dict(p, max_path_length)
+                for p in env_infos
+            ])
+
+            valids = [np.ones_like(path['returns']) for path in paths]
+            valids = tensor_utils.pad_tensor_n(valids, max_path_length)
+
+            average_discounted_return = \
+                np.mean([path['returns'][0] for path in paths])
+
+            undiscounted_returns = [sum(path['rewards']) for path in paths]
+
+            ent = np.sum(
+                self.policy.distribution.entropy(agent_infos) *
+                valids) / np.sum(valids)
+
+            samples_data = dict(
+                observations=obs,
+                actions=actions,
+                advantages=adv,
+                rewards=rewards,
+                returns=returns,
+                valids=valids,
+                agent_infos=agent_infos,
+                env_infos=env_infos,
+                paths=paths,
+            )
+
+        logger.log('fitting baseline...')
+        if hasattr(self.baseline, 'fit_with_samples'):
```

Comment:
Why do we fit baseline in preprocess?

Reply:
this was probably just copied from old code, but yes it should be fixes.